runtime/src/lib.rs

```diff
@@ -752,6 +1003,115 @@ pub type Executive = frame_executive::Executive<
+						return false;
+					};
+
+					let (Some(first_char_a1), Some(first_char_a2)) = (a1.chars().next(), a2.chars().next()) else {
+						return false;
+					};
+
+					(first_char_a1.is_alphabetic() && first_char_a2.is_alphabetic()) // ensure first character is an alphabet
+						&&
+						(a1 == a1.to_lowercase() && a2 == a2.to_lowercase()) // ensure lowercase
+						&&
+						a1 <= a2 // ensure sorted. Note: "Strings are ordered lexicographically by their byte values ... This is not necessarily the same as “alphabetical” order, which varies by language and locale". Source: https://doc.rust-lang.org/std/primitive.str.html#impl-Ord-for-str
+						&&
+						a1 != a2 // ensure no duplicates
+				})
+			},
+		},
+		Categories::Shards(shards_script_info_struct) => {
+			let format = shards_script_info_struct.format;
+			let requiring = &shards_script_info_struct.requiring;
+			let implementing = &shards_script_info_struct.implementing;
+
+			let all_required_trait_impls_found = requiring.iter().all(|shards_trait| {
+				proto_references.iter().any(|proto| {
+					if let Some(trait_impls) = pallet_protos::TraitImplsByShard::<Runtime>::get(proto) {
+						trait_impls.contains(shards_trait)
+					} else {
+						false
+					}
+				})
+			});
+
+			let all_traits_implemented_in_this_shards = implementing.iter().all(|shards_trait| {
+				match format {
+					ShardsFormat::Edn => {
+						// TODO - How do I check if this Shards is implementing this Trait - Giovanni Petrantoni
+						false
+					},
+					ShardsFormat::Binary => {
+						// TODO - How do I check if this Shards is implementing this Trait - Giovanni Petrantoni
+						false
+					},
+				}
+			});
+
+			all_required_trait_impls_found && all_traits_implemented_in_this_shards
+		},
+		Categories::Audio(sub_categories) => match sub_categories {
+			AudioCategories::OggFile => infer::is(data, "ogg"),  // TODO Review - We are not checking for other OGG file extensions https://en.wikipedia.org/wiki/Ogg
+			AudioCategories::Mp3File => infer::is(data, "mp3"),
+		},
+		Categories::Texture(sub_categories) => match sub_categories {
+			TextureCategories::PngFile => infer::is(data, "png"), // png_decoder::decode(&data[..]).is_ok(),
+			TextureCategories::JpgFile => infer::is(data, "jpg"), // TODO - Review because this does not include ".jpeg" images, only ".jpg" images
+		},
+		Categories::Vector(sub_categories) => match sub_categories {
+			VectorCategories::SvgFile => false, // TODO - Giovanni Petrantoni
+			VectorCategories::TtfFile => infer::is(data, "ttf"), // ttf_parser::Face::parse(&data[..], 0).is_ok(),
+		},
+		Categories::Video(sub_categories) => match sub_categories {
+			VideoCategories::MkvFile => infer::is(data, "mkv"),
+			VideoCategories::Mp4File => infer::is(data, "mp4"),
+		},
+		Categories::Model(sub_categories) => match sub_categories {
+			ModelCategories::GltfFile => false, // TODO - Giovanni Petrantoni
```

Comment:
@sinkingsugar - what do we do about all file types whose file signature / magic number (https://en.wikipedia.org/wiki/List_of_file_signatures) **does not exist in the `infer` crate?**

Should I manually verify them (i.e search online for their magic number and verify it against `data`) or just leave them as `false` - since anyways this is literally just a band-aid solution that won't be used for the real thing 😅?

(Btw all the file types that can't be verified using `infer` crate are labelled as `// TODO - Giovanni Petrantoni` 😅 )

Reply:
Don't need to put my name lol